### PR TITLE
Fixes for failures when using astropy develop.

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Currently, `poppy` is causing failures in the `jwst` development dependencies CI. These failures are due to a change in the some of the APIs in `astropy` introduced by the merger of PR astropy/astropy#12633. This PR resolves these issues, allowing `poppy` to properly work with the current state of the `astropy` develop branch.

### CODE REVIEW

**Comment by nden** [comment]
@mperrin @shanosborne I am not sure who should review this PR. 
Could you assign someone? 
We'd appreciate a quick resolution and a release, as we are about to make a jwst release in a week.

**Comment by mperrin** [comment]
@nden Will do! @shanosborne are you able to review this for a quick turnaround?

**Comment by WilliamJamieson** [comment]
The test failures are likely due to the numpy 1.22 changes, see numpy/numpy#19478. These changes can cause minor numerical, which may require updating the tolerances used for `np.allclose` tests.

It maybe possible to disable some cpu features instead, see astropy/astropy#12684 for details.

**Review by nden — APPROVED** [review]

**Comment by nden** [comment]
Ping to get attention to this issue.
For what it's worth it appears I have permissions to review it so I did.

**Comment by mperrin** [comment]
Hi @nden and @WilliamJamieson - thanks for the ping. @shanosborne said she's available to take a look at this today. (She was out last week; thanks for your patience)

**Review by shanosborne — APPROVED** [review]
This change fixes the astropy develop issue and the `test_rotations` failures in the CI are unrelated to this PR so I can fix them separately and not hold this PR up. This is good to go.

**Pull request merged**
